app/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersPartnershipConfirmRegisteredAddressController.scala

```diff
@@ -17,47 +17,64 @@
 package uk.gov.hmrc.fhregistrationfrontend.controllers
 
 import com.google.inject.{Inject, Singleton}
-import play.api.mvc.{Action, AnyContent, Call, MessagesControllerComponents, Results}
+import models.{Mode, NormalMode}
+import play.api.mvc.{Action, AnyContent, Call, MessagesControllerComponents}
 import uk.gov.hmrc.fhregistrationfrontend.actions.Actions
-import uk.gov.hmrc.fhregistrationfrontend.config.FrontendAppConfig
-import uk.gov.hmrc.fhregistrationfrontend.forms.models.Address
+import uk.gov.hmrc.fhregistrationfrontend.pages.businessPartners.{AddressPage, UkAddressLookupPage}
+import uk.gov.hmrc.fhregistrationfrontend.repositories.SessionRepository
 import uk.gov.hmrc.fhregistrationfrontend.views.Views
 
+import scala.concurrent.{ExecutionContext, Future}
+
 @Singleton
 class BusinessPartnersPartnershipConfirmRegisteredAddressController @Inject()(
   ds: CommonPlayDependencies,
   view: Views,
   actions: Actions,
-  config: FrontendAppConfig)(
+  val sessionCache: SessionRepository)(
   cc: MessagesControllerComponents
-) extends AppController(ds, cc) {
+)(implicit val ec: ExecutionContext)
+    extends AppController(ds, cc) with ControllerHelper {
   import actions._
 
-  val postAction: Call = routes.BusinessPartnersPartnershipConfirmRegisteredAddressController.next()
+  def postAction(index: Int, mode: Mode): Call =
+    routes.BusinessPartnersPartnershipConfirmRegisteredAddressController.next(index)
+  def editAddressUrl(index: Int): String =
+    routes.BusinessPartnersPartnershipEnterAddressController.load(index, NormalMode).url
+  val journey = "partnership"
+  val backLink = "#"
```

Comment:
It looks like the 'Save and continue' navigation has been added as part of this ticket. Is it worth also adding the 'Back' button nav with this ticket so that it doesn't get missed? If not, we should probably get a new ticket made for that.